src/Telemetry/Opt_In/Status.php

```diff
@@ -238,11 +238,18 @@ public function set_status( bool $status, string $stellar_slug = '' ) {
 
 		$option = $this->get_option();
 
+		// Set the sent plugin's optin status.
 		$option['plugins'][ $stellar_slug ]['optin'] = $status;
 
+		// Force the others to all match.
+		foreach ( $option['plugins'] as &$plugin_data ) {
```

Comment:
At [this line](https://github.com/stellarwp/telemetry/pull/100/files#diff-8be681e9c0b0e6e34aad26df03f9ab351b19cdddb5d98176e20e36a9c6a1fc83R76) you've added a check to make sure the `$option['plugins']` will default to an empty array if not set; the same possibility of the `$option['plugins']` variable not being set should be handled here:
`if( isset( $option['plugins'])) { foreach ... }